libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -539,6 +572,16 @@ private void initializeRoute(MapView mapView, MapboxMap map) {
     mapRoute = new NavigationMapRoute(null, mapView, map, routeStyleRes);
   }
 
+  private void initializeCamera(MapboxMap map) {
+    mapCamera = new NavigationCamera(map, locationComponent);
+  }
+
+  private void initializeFpsDelegate(MapView mapView) {
+    MapBatteryMonitor batteryMonitor = new MapBatteryMonitor();
+    mapFpsDelegate = new MapFpsDelegate(mapView, batteryMonitor);
```

Comment:
`MapFpsDelegate` can implement `TrackingModeTransitionListener` directly instead of the nested `FpsDelegateTrackingChangedListener` and can be added to/removed from the `mapCamera`'s listeners list in `NavigationMapboxMap`'s `onStart`/`onStop`. This will decrease the coupling between `MapFpsDelegate` and `NavigationCamera` as one will not depend on the other anymore.

Reply:
I believe that the same reasoning can be applied to the `FpsDelegateProgressChangeListener`.

Reply:
This is a great recommendation - no need to create internal listeners as the class is package-private.  For now, I only reworked the `MapFpsDelegate` / `NavigationCamera`.  

> I believe that the same reasoning can be applied to the `FpsDelegateProgressChangeListener`.  

☝️ This requires some bigger API rework (as most of the classes in `NavigationMapboxMap` follow the same setup) that I'd like to revisit in a later PR.